transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java

```diff
@@ -434,14 +432,16 @@ private String generateModel(String filePath,
             return getJavaQualifiedName(filePath, monitor);
         }
         case JSON: {
+            String className = classNameBuilder.toString().replace("_", "");
```

Comment:
"The 3rd-party library we're using is removing the underscore from the name during code generation. Not sure what other characters might be treated the same way, or how to even determine this. It will just take some experimentation I think"
we can have a look to source code of used library which seems to be https://github.com/joelittlejohn/jsonschema2pojo or by experimentation we can write a paraterized test providing all (java or UTF-8 or ??) possible characters and checking for the output